core/src/main/java/io/questdb/client/Sender.java

```diff
@@ -489,6 +492,42 @@ public LineSenderBuilder autoFlushRows(int autoFlushRows) {
             return this;
         }
 
+
+        /**
+         * Set the interval at which the Sender automatically flushes its buffer.
+         * <br>
+         * It flushed the buffer even when the number of buffered rows is less than the value set by {@link #autoFlushRows(int)}.
+         * This prevents rows from being locally buffered for too long when the rate of incoming data is low.
+         * <p>
+         * <strong>Important:</strong>This option does not cause the Sender to flush the buffer at regular intervals.
+         * Auto-flushing is only triggered when calling {@link #atNow()}, {@link #at(Instant)} or {@link #at(long, ChronoUnit)}.
+         * The Sender will not flush the buffer if no new rows are added even if the auto-flush interval has elapsed.
+         * <p>
+         * This is only used when communicating over HTTP transport, and it's illegal to call this method when
+         * communicating over TCP transport.
+         * <br>
+         * You cannot set this value when auto-flush is disabled. See {@link #disableAutoFlush()}.
+         * <br>
+         * Default value is 1,000 milliseconds.
+         *
+         * @param autoFlushIntervalMillis interval at which the Sender automatically flushes its buffer in milliseconds.
+         * @return this instance for method chaining
+         */
+        public LineSenderBuilder autoFlushIntervalMillis(int autoFlushIntervalMillis) {
```

Comment:
Adding context here, rather than thinking anything needs to change.

In the Python client one will be able to select the three auto flushing modes independent of each other. Here's the logic so far:

```cython
cdef bint _should_auto_flush(
            const auto_flush_t* af,
            line_sender_buffer* ls_buf,
            int64_t last_flush_ms):
    if not af.mode.enabled or (af.sender == NULL):
        return False

    # Check for interval breach.
    if (af.mode.interval != -1) and \
        (((line_sender_now_micros() / 1000) - last_flush_ms) >= af.mode.interval):
        return True

    # Check `auto_flush_rows` breach.
    if (af.mode.row_count != -1) and \
        (line_sender_buffer_row_count(ls_buf) >= af.mode.row_count):
        return True

    # Check `auto_flush_bytes` breach.
    if (af.mode.byte_count != -1) and \
        (line_sender_buffer_row_count(ls_buf) >= af.mode.byte_count):
        return True

    return False
```

Auto-flushing will be consistent in Python across protocols.

Reply:
I see. the differences stem from this: in Java HTTP and TCP transports behave differently. If TCP is here to stay then it might make sense to unify the behaviour.